src/app/models/Annotation.ts

```diff
@@ -0,0 +1,148 @@
+import { Injector } from "@angular/core";
+import {
+  AUDIO_EVENT_TAG,
+  AUDIO_RECORDING,
+  SHALLOW_AUDIO_EVENT,
+  SHALLOW_SITE,
+  TAG,
+} from "@baw-api/ServiceTokens";
+import { Id } from "@interfaces/apiInterfaces";
+import { BehaviorSubject, Observable } from "rxjs";
+import { flatMap, map } from "rxjs/operators";
+import { AbstractModel } from "./AbstractModel";
+import type { AudioEvent } from "./AudioEvent";
+import type { AudioEventTag } from "./AudioEventTag";
+import type { AudioRecording } from "./AudioRecording";
+import type { Site } from "./Site";
+import type { Tag } from "./Tag";
+
+export interface IAnnotation {
+  id: Id;
+  audioEvent: Observable<AudioEvent>;
+  audioEventTags: Observable<AudioEventTag[]>;
+  tags: Observable<Tag[]>;
```

Comment:
I don't think it's necessary to have both of these.

I'd suggest keeping audioEventTags since that model had meta information about when the tag was applied to the event.

Reply:
`AudioEventTag` only gives the following values:

- id
- audioEventId
- tagId
- creatorId/At
- updaterId/At

I need the tag text a lot of the time. Is there something im missing?

Reply:
There are different use cases.

If you're showing annotations on some summary page then: `audioEvent.audioEventTags.select(aet => aet.tag.text)` is approproate.

If you're editing an anntoation, you don't need text at all: `audioEvent.audioEventTags.push( /*new value*/)`

Regardless both are associations (one is transitive) of the audio event.